src/frontend/src/session.rs

```diff
@@ -310,17 +303,31 @@ impl FrontendEnv {
         self.meta_client.clone()
     }
 
-    pub fn query_manager(&self) -> &QueryManager {
-        &self.query_manager
+    pub fn hummock_snapshot_manager(&self) -> &HummockSnapshotManager {
+        &*self.hummock_snapshot_manager
     }
 
-    pub fn hummock_snapshot_manager(&self) -> &HummockSnapshotManagerRef {
-        &self.hummock_snapshot_manager
+    pub fn hummock_snapshot_manager_ref(&self) -> HummockSnapshotManagerRef {
+        self.hummock_snapshot_manager.clone()
     }
 
     pub fn server_address(&self) -> &HostAddr {
         &self.server_addr
     }
+
+    pub async fn query_manager(&self) -> &QueryManager {
```

Comment:
It's really wierd to me to introduce circular dependency here  😂.

Reply:
`QueryManager` have `Ref` to fields in `FrontendEnv`, and also has `context`, i.e. a `FrontendEnv`, as the function parameters when someone calls `schedule` and `schedule_single`.

My intention is to remove one of them so they won't mismatch with each other, I am totally fine with other approaches

Calling with `context` seems a bit error-prone unless we make `QueryManager` stateless, 

open to discussion

Reply:
I prefer to remove the `context` in `QueryManager`'s `schedule` methods.

Reply:
But I don't want store whole `FrontedEnv` in `QueryManager` since this introduces two problems:
1. Add unnecessary depencies to `QueryManager`
2. Introduce circular dependencies.